splice/web/api/content.py

```diff
@@ -0,0 +1,98 @@
+from flask import Blueprint, request
+from flask.json import jsonify
+from splice.web.api.content_upload import upload_signed_content, resign_content
+from splice.queries.common import session_scope
+from splice.queries.content import get_content, get_contents, insert_content, update_content
+
+
+content_bp = Blueprint('api.content', __name__, url_prefix='/api')
+
+
+@content_bp.route('/content/resignall', methods=['POST'])
+def handler_content_resign_all():
+    succeeded, failed = [], []
+    for content in get_contents():
+        try:
+            resign_content(content['name'], content['version'])
+        except Exception as e:
+            failed.append("%s: %s" % (content['name'], e))
+        else:
+            succeeded.append(content['name'])
+
+    return jsonify(succeeded=succeeded, failed=failed)
+
+
+@content_bp.route('/content/sign', methods=['POST'])
+def handler_content_upload():
+    """Signed and upload a piece of content to S3, return the URL if succeeds."""
+    def _is_allowed_file(name):
+        return '.' in name and name.rsplit('.', 1)[1].lower() in set(['zip'])
```

Comment:
Yup, let's try libmagic then.

For the xpi, since it's already zipped as a single archive, how should we embed the manifest.json. I naively thought we might have to sign it as a single file.